### PR TITLE
Changes made for dynamic selector

### PR DESCRIPTION
**Pull request by lxcleo**
Added codes in dynamic base to allow dynamic selector work with orientation. Also changed GetActiveModel function to public for use.

### CODE REVIEW

File: Core/Models/DynamicsBase.hpp [hunk 1]
```diff
@@ -108,6 +108,11 @@ class DynamicsBase{
     [[nodiscard]] virtual inline bool DynamicsArePaused(void) const{
         return dynamics_paused_;
     }
+    // Calling GetPosition returns the quaternion as a Vector4
+    [[nodiscard]] virtual inline Quaternion GetOrientation(void) const
+    {
+        return Quaternion(position_.data());
+    }
```
Comment (jianwei-sun) [inline Core/Models/DynamicsBase.hpp:115]: I am hesitant about adding this member function to the base class, since then it would imply that any dynamics object has an orientation. (E.g., if we're using this for a 5 dof joint space admittance control, what would the orientation even mean?) This member function only makes sense when `DynamicsBase` is an `RotationSecondOrder`, so adding it here feels too restrictive. 

I understand your reasoning for doing this though. I think in the VRex code, you could instead do something like:
```
Eigen::Quaternion(model.GetPosition().data());
```
and this would be equivalent and would not require modifications to the base class.

File: Core/Containers/DynamicsSelector.hpp [hunk 2]
```diff
@@ -129,21 +135,28 @@ class DynamicsSelector : public DynamicsBase<
...
             return &std::get<I>(models_);
-        } else if constexpr(I + 1 < std::tuple_size_v<std::tuple<Models...>>){
+        }
+        else if constexpr (I + 1 < std::tuple_size_v<std::tuple<Models...>>)
+        {
             return GetActiveModel<I + 1>();
-        } else{
+        }
+        else
+        {
```
Comment (jianwei-sun) [inline Core/Containers/DynamicsSelector.hpp:154]: The auto formatter looks like it changed these lines. Please follow the existing formatting

File: Core/Containers/DynamicsSelector.hpp [hunk 3]
```diff
@@ -129,21 +135,28 @@ class DynamicsSelector : public DynamicsBase<
...
             return nullptr;
         }
     }
 
+    // private:
     template <size_t I = 0>
     const Base* GetActiveModel() const{
```
Comment (jianwei-sun) [inline Core/Containers/DynamicsSelector.hpp:161]: The `private` keyword should be moved down instead of commented out, since now the member variables (`models_` and `index_`) are `public`.

File: Core/Containers/DynamicsSelector.hpp [hunk 4]
```diff
@@ -68,6 +69,11 @@ class DynamicsSelector : public DynamicsBase<
     [[nodiscard]] bool DynamicsArePaused(void) const override{
         return GetActiveModel()->DynamicsArePaused();
     }
+    // Calling GetPosition returns the quaternion as a Vector4
+    [[nodiscard]] inline Quaternion GetOrientation(void) const
+    {
+        return GetActiveModel()->GetOrientation();
+    }
```
Comment (jianwei-sun) [inline Core/Containers/DynamicsSelector.hpp:76]: Please see comment in `DynamicsBase`.

**Review by jianwei-sun — CHANGES_REQUESTED** [review]
Thanks for opening this pull request. I left some comments below.

File: Core/Containers/DynamicsSelector.hpp (hunk 2 above)
Comment (lxcleo) [reply Core/Containers/DynamicsSelector.hpp]: The format has been switch to the existing formatting

File: Core/Containers/DynamicsSelector.hpp (hunk 4 above)
Comment (lxcleo) [reply Core/Containers/DynamicsSelector.hpp]: The GetOrientation function has been removed from DynamicsSelector

File: Core/Models/DynamicsBase.hpp [hunk 5]
```diff
@@ -33,7 +33,6 @@ class DynamicsBase{
 
     using PositionBound = BoundBase<PositionDimensions, Scalar>;
     using PositionBoundPtr = std::shared_ptr<PositionBound>;
-
     const static unsigned int Dimension = Dimensions;
```
Comment (jianwei-sun) [inline Core/Models/DynamicsBase.hpp:36]: To undo the changes to this file, you can just cherry-pick it from `main`.

File: Core/Containers/DynamicsSelector.hpp [hunk 6]
```diff
@@ -143,7 +140,10 @@ class DynamicsSelector : public DynamicsBase<
...
         }
     }
-
+private:
+    // Helper function for retrieving the currently active model at runtime, by indexing the tuple
+    // using template recursion. Since all models inherit from the same base type under this class,
+    // their pointers can be casted to the base type
     template <size_t I = 0>
     const Base* GetActiveModel() const{
         if(I == index_){
```
Comment (jianwei-sun) [inline Core/Containers/DynamicsSelector.hpp:149]: The `const` version of this function will probably also need to be in `public`

Comment (lxcleo) [reply Core/Containers/DynamicsSelector.hpp]: Moved the private statement down to allow (models_ and index_) variables to be private, and const version of GetActiveModel() to be public.

File: Core/Models/DynamicsBase.hpp (hunk 5 above)
Comment (lxcleo) [reply Core/Models/DynamicsBase.hpp]: Cherry-picked the function from main
